File: stat/src/main/java/com/hstat/kafka/TgbConsumer.java
package com.hstat.kafka;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.hstat.dto.StatSend;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class TgbConsumer {

    private static final String orderTopic = "${topic.name}";
    private final ObjectMapper objectMapper;
    @Autowired
    public TgbConsumer(ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }

    @KafkaListener(topics = orderTopic)
    public void consumeMessage(String message) throws JsonProcessingException {

        StatSend received = objectMapper.readValue(message, StatSend.class);
        System.out.println(received);
    }

}

File: stat_kt/src/main/java/com/ktstat/kafka/TgbConsumer.kt
package com.ktstat.kafka

import com.fasterxml.jackson.databind.ObjectMapper
import com.hstat.common.CommonConstantsKt
import com.hstat.common.dtoModels.StatSend
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component


@Component
class TgbConsumer (private val objectMapper: ObjectMapper){


    @KafkaListener(topics = [CommonConstantsKt.TG_STAT_TOPIC])
    fun consumeMessage(message: String){
        val received = objectMapper.readValue(message, StatSend::class.java)
        println(received)
    }
}

